eng/validation/test-publishing.ps1

```diff
@@ -56,3 +80,14 @@ if(-not $postCheck)
 {
     Write-Error "Build was not added to '${global:targetChannel}'."
 }
+
+## Clean up branch if successful
+Write-Host "Build was successful. Cleaning up ${global:targetBranch} branch."
+try
+{
```

Comment:
In this case I don't think leaving the branch around for investigation is all that interesting. I think we should clean up regardless of success.